lib/pavilion/plugins/commands/wait.py

```diff
@@ -49,8 +56,7 @@ def run(self, pav_cfg, args):
 
         final_statuses = 0
 
-        while (final_statuses < len(tmp_statuses)) and \
-              ((time.time() - start_time) < float(args.timeout)):
+        while final_statuses < len(tmp_statuses):
```

Comment:
This would be greatly simplified by defaulting --timeout to None (no timeout), and dropping the no_timeout argument.

```python
end_time = None
if args.timeout is not None:
    end_time = time.time() + args.timeout
while (final_statuses < len(tmp_statuses) and
        (end_time is None or time.time() < end_time)):
```